src/main/scala/org/constellation/rollback/RollbackLoader.scala

```diff
@@ -0,0 +1,56 @@
+package org.constellation.rollback
+
+import better.files.File
+import com.typesafe.scalalogging.Logger
+import org.constellation.consensus.{SnapshotInfo, StoredSnapshot}
+import org.constellation.primitives.Schema.GenesisObservation
+import org.constellation.serializer.KryoSerializer
+import org.slf4j.LoggerFactory
+
+import scala.util.{Failure, Success, Try}
+
+class RollbackLoader(
+  rollbackDataFolder: String = "rollback_data",
+  snapshotDataFolder: String = "rollback_data/snapshots",
+  snapshotInfoFile: String = "rollback_info",
+  genesisObservationFile: String = "rollback_genesis"
+) {
+
+  val logger = Logger(LoggerFactory.getLogger(getClass.getName))
+
+  def loadSnapshotsFromFile(): Either[RollbackLoaderException, Seq[StoredSnapshot]] =
```

Comment:
In my opinion, every I/O operation (such as operations on files) should be wrapped into i.e. IO monad, which will postpone the evaluation of the operations. You can use either `IO`/`F: Concurent` or combined with Resource/Bracket which will be the exact file